src/components/permissions/rowLevelPolicyHelpers.ts

```diff
@@ -167,18 +176,33 @@ const compileRowLevelPolicy = (
   usingCondition: string,
   condition: string
 ) => {
+  // Create function for permission so that row level policies are checked independently
+  // SECURITY DEFINER is used to make sure permission check does not take into account existing permissions
+  // i.e. without this function, if we have something like this "restrict review by review_assignment where user is reviewer"
+  // and then we have another restriction of "review_assignment where user is assigner" in the former policy check the latter check will also run
+  const functionName = `private.policy_check_${permissionAbbreviation}_${tableName}`
+  const permissionFunction = `
+    CREATE OR REPLACE FUNCTION ${functionName} ()
+      RETURNS SETOF ${tableName}
+      SECURITY DEFINER
+      AS $$
+        SELECT * FROM ${tableName} WHERE ${usingCondition}
+  $$
+  LANGUAGE sql
+  STABLE;
+  `
+
   // We want to first check if the permission exists in JWT token (to make sure we only do the full query if it exists),
   // so we add a check for "policy": true, i.e. jwt_get_boolean('pp2pn2tp2') = true
   // this will become first condition in both USING and WITH CHECK of every row level rule
   const addBracketsAndPermissionCheck = (condition: string) =>
-    `(COALESCE(current_setting('jwt.claims.${permissionAbbreviation}', true),'') != '' and ${condition})`
+    `(COALESCE(current_setting('jwt.claims.${permissionAbbreviation}', true),'') != '' AND ${condition})`
 
-  // If using clause is not used return nothing
-  // otherwise return USING from usingCondition (view rule, see compileRowLevelPolicies) or condition (current rule condition)
+  // If USING clause is not used return nothing
+  // otherwise return USING from usingCondition (which comes from the defined 'view' rule))
   const getUsingClause = () => {
     if (!ruleSettings.using) return ''
-    const conditionToUse = ruleSettings.withCheck ? usingCondition : condition
-    return `USING ${addBracketsAndPermissionCheck(conditionToUse)}`
```

Comment:
Logic here is slightly changed, but usingCondition is always available (if view permissions are configured in rules, see https://github.com/msupply-foundation/conforma-server/blob/028193d5d11e2ae9b30484c108e9d11ac642f64d/src/components/permissions/rowLevelPolicyHelpers.ts#L133-L137